embulk-input-s3/src/main/java/org/embulk/input/s3/AlwaysRetryable.java

```diff
@@ -155,4 +179,29 @@ public T executeWith(RetryExecutor executor)
             throw Throwables.propagate(e);
         }
     }
+
+    private static boolean isAse(SdkBaseException e)
+    {
+        return e instanceof AmazonServiceException;
+    }
+
+    /**
+     * Returns true if the specified exception is a non-retryable service side exception.
+     *
+     * @param exception The exception to test.
+     * @return True if the exception resulted from a non-retryable service error, otherwise false.
+     */
+    public static boolean isNonRetryableServiceException(SdkBaseException exception)
+    {
+        if (!isAse(exception)) {
```

Comment:
I think a ClassCastException would would already happened at the caller side https://github.com/embulk/embulk-input-s3/pull/74/files#diff-6c8887c59723a08b34aa4f0e2976aefaR86 if this is not a AmazonExceptionService?